tests/realtikvtest/addindextest1/disttask_test.go

```diff
@@ -61,6 +77,18 @@ func TestAddIndexDistBasic(t *testing.T) {
 	tk.MustExec("split table t between (3) and (8646911284551352360) regions 50;")
 	tk.MustExec("alter table t add index idx(a);")
 	tk.MustExec("admin check index t idx;")
+	taskMgr, err := storage.GetTaskManager()
+	require.NoError(t, err)
+	ctx := util.WithInternalSourceType(context.Background(), "dispatcher")
+	task, err := taskMgr.GetTaskByIDWithHistory(ctx, storage.TestLastTaskID.Load())
+	require.NoError(t, err)
+	require.Equal(t, 1, task.Concurrency)
+	taskMeta := ddl.BackfillTaskMeta{}
+	require.NoError(t, json.Unmarshal(task.Meta, &taskMeta))
+	require.Equal(t, 111, taskMeta.WorkerCntLimit)
```

Comment:
Why setting workerCntLimit > cpu count? If task concurrency != workerCnt, the reserved resource is less than real consuming resource.

Reply:
1 add-index worker might not consume 1 core, so `GetDDLReorgWorkerCounter` can be larger than core-count, see https://docs.pingcap.com/tidb/stable/system-variables#tidb_ddl_reorg_worker_cnt.

if user do set tidb_ddl_reorg_worker_cnt > core-count, we reserve resource using core-count, and that's what we can reserve the most.